cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -420,6 +419,259 @@ class self_comparator {
 };
 
 }  // namespace lexicographic
+
+namespace equality {
+
+template <typename Nullate>
+class device_row_comparator {
+  friend class self_comparator;
+
+ public:
+  /**
+   * @brief Checks whether the row at `lhs_index` in the `lhs` table is equal to the row at
+   * `rhs_index` in the `rhs` table.
+   *
+   * @param lhs_index The index of row in the `lhs` table to examine
+   * @param rhs_index The index of the row in the `rhs` table to examine
+   * @return `true` if row from the `lhs` table is equal to the row in the `rhs` table
+   */
+  __device__ bool operator()(size_type const lhs_index, size_type const rhs_index) const noexcept
+  {
+    auto equal_elements = [=](column_device_view l, column_device_view r) {
```

Comment:
Minor nit: Is there any reason to assign this to a temporary? I feel like I usually see lambdas for algorithms constructed as temporaries passed as function parameters.

Pedantic (apologies in advance if you're already aware of all this): default captures (even by-value captures) can be dangerous and lead to subtle bugs. This lambda in fact demonstrates the textbook case where there's an implicit capture by reference: you're capturing `nulls` and `nulls_are_equal`, which are class members, and therefore the lambda is implicitly capturing `this`. I don't see any real problems in this case though(hence pedantic).

Reply:
No particular reason. This came verbatim from https://github.com/rapidsai/cudf/blob/48cebf7ac498439ebae3d5a50581a552f8fed88d/cpp/include/cudf/table/row_operators.cuh#L244-L251 I assume it was a readability thing. Some might prefer seeing a descriptive name for the temporary lambda.